src/Plugin/PasswordConstraint/UpperCase.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Drupal\password_enhancements\Plugin\PasswordConstraint;
+
+/**
+ * Upper-case password constraint plugin.
+ *
+ * @PasswordConstraint(
+ *   id = "upper_case",
+ *   name = @Translation("Upper-case"),
+ *   description = @Translation("Checks if the password has at least a specified number of upper-cased character."),
+ *   unique = TRUE,
+ *   jsLibrary = "password_enhancements/plugin.upper_case",
+ * )
+ */
+final class UpperCase extends MinimumCharacters {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultDescriptionSingular(): string {
+    return 'Add at least one upper-cased letter.';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultDescriptionPlural(): string {
+    return 'Add @minimum_characters more upper-cased letters.';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate(string $value): bool {
+    return parent::validate(preg_replace('/([^A-Z])/', '', $value));
```

Comment:
I think it is probably better to let it fail in case of null instead of letting that through or casting it to an empty string if it is null, but since $value is always string it should be okay.